cinder/backup/manager.py

```diff
@@ -416,41 +417,16 @@ def create_backup(self, context, backup):
                          'previous_status': 'error_backing-up'})
                 self._update_backup_error(backup, six.text_type(err))
 
-        # Restore the original status.
-        if snapshot_id:
-            self.db.snapshot_update(
-                context, snapshot_id,
-                {'status': fields.SnapshotStatus.AVAILABLE})
-        else:
-            self.db.volume_update(context, volume_id,
-                                  {'status': previous_status,
-                                   'previous_status': 'backing-up'})
+    @utils.trace
+    def _start_backup(self, context, backup, volume):
+        """This starts the backup process.
 
-        # _run_backup method above updated the status for the backup, so it
-        # will reflect latest status, even if it is deleted
-        completion_msg = 'finished'
-        if backup.status in (fields.BackupStatus.DELETING,
-                             fields.BackupStatus.DELETED):
-            completion_msg = 'aborted'
-        else:
-            backup.status = fields.BackupStatus.AVAILABLE
-            backup.size = volume['size']
-
-            if updates:
-                backup.update(updates)
-            backup.save()
+           First we have to get the backup device from the volume manager.
+           This can take a long time to complete.  Once the volume manager
+           is done creating/getting the backup device, then we get a callback
```

Comment:
imho, this line sounds better without the `then`

Reply:
done